### DISCUSSION TITLE
Can't change volume

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#120] [Fixed Issues] [answered]
@rern 

RPi 3B+, Merus Audi Amp rAudio 1, Update 2020015 applied

Since applying the 20200115 update I cannot change the volume on this installation, fortunately it is still set to a reasonable level.  When I enter the Playback screen the Volume level shows 100 and despite any changes to the Volume level, which don't seem to do anything, whenever I exit this screen and return it reverts to 100. In MPD:

Audio Output Device:  Merus Audio Amp
Mixer Device: A.Mstr Vol (set to 100 under small speaker icon)
Volume Control: Mixer Device
Bit Perfect No Volume: Off
FFMpeg Decoder: On

All other options default / off

Chris

**Comment by findnix** [comment]
RPi 4B,  Pre Box S2 Digital, rAudio 1, Update 2020015 

same here:
no change in volume.
volume dial jumps back to last successfully changes volume
i can change volume with alsamixer.
changing volume with slider in "settings-> mpd -> output -> mixerdevice" works.
```
Jan 17 17:38:31 rAudio dbus-daemon[284]: [system] Activating via systemd: service name='org.freedesktop.home1' unit='dbus-org.freedesktop.home1.service' requested by ':1.295' (uid=0 pid=11103 comm="/usr/bin/sudo /srv/http/bash/cmd.sh volume?63?95?P")
Jan 17 17:38:31 rAudio dbus-daemon[284]: [system] Activation via systemd failed for unit 'dbus-org.freedesktop.home1.service': Unit dbus-org.freedesktop.home1.service not found.
Jan 17 17:38:31 rAudio sudo[11103]: pam_systemd_home(sudo:account): systemd-homed is not available: Unit dbus-org.freedesktop.home1.service not found.
Jan 17 17:38:31 rAudio audit[11103]: USER_ACCT pid=11103 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:accounting grantors=pam_unix,pam_permit,pam_time acct="http" exe="/usr/bin/sudo" hostname=? addr=? terminal=? res=success'
Jan 17 17:38:31 rAudio audit[11103]: CRED_REFR pid=11103 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=pam_faillock,pam_permit,pam_env,pam_faillock acct="root" exe="/usr/bin/sudo" hostname=? addr=? terminal=? res=success'
Jan 17 17:38:31 rAudio audit[11103]: USER_START pid=11103 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:session_open grantors=pam_limits,pam_unix,pam_permit acct="root" exe="/usr/bin/sudo" hostname=? addr=? terminal=? res=success'
Jan 17 17:38:31 rAudio sudo[11103]:     http : PWD=/srv/http ; USER=root ; COMMAND=/srv/http/bash/cmd.sh volume
                                    63
                                    95
                                    Pre Box S2 Digital
Jan 17 17:38:31 rAudio sudo[11103]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=33)
Jan 17 17:38:33 rAudio sudo[11103]: pam_unix(sudo:session): session closed for user root

```
```
Jan 17 17:35:37 rAudio dbus-daemon[284]: [system] Activating via systemd: service name='org.freedesktop.home1' unit='dbus-org.freedesktop.home1.service' requested by ':1.285' (uid=0 pid=10724 comm="/usr/bin/sudo /usr/bin/amixer -M sset Pre Box S2 D")
Jan 17 17:35:37 rAudio dbus-daemon[284]: [system] Activation via systemd failed for unit 'dbus-org.freedesktop.home1.service': Unit dbus-org.freedesktop.home1.service not found.
Jan 17 17:35:37 rAudio sudo[10724]: pam_systemd_home(sudo:account): systemd-homed is not available: Unit dbus-org.freedesktop.home1.service not found.
Jan 17 17:35:37 rAudio audit[10724]: USER_ACCT pid=10724 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:accounting grantors=pam_unix,pam_permit,pam_time acct="http" exe="/usr/bin/sudo" hostname=? addr=? terminal=? res=success'
Jan 17 17:35:37 rAudio audit[10724]: CRED_REFR pid=10724 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=pam_faillock,pam_permit,pam_env,pam_faillock acct="root" exe="/usr/bin/sudo" hostname=? addr=? terminal=? res=success'
Jan 17 17:35:37 rAudio sudo[10724]:     http : PWD=/srv/http ; USER=root ; COMMAND=/usr/bin/amixer -M sset Pre Box S2 Digital  64%
Jan 17 17:35:37 rAudio audit[10724]: USER_START pid=10724 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:session_open grantors=pam_limits,pam_unix,pam_permit acct="root" exe="/usr/bin/sudo" hostname=? addr=? terminal=? res=success'
Jan 17 17:35:37 rAudio sudo[10724]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=33)
Jan 17 17:35:37 rAudio sudo[10724]: pam_unix(sudo:session): session closed for user root
Jan 17 17:35:37 rAudio sudo[10723]: pam_unix(sudo:session): session closed for user root

```

**Comment by ceejayemm** [comment]
@rern 
Based on the info from @findnix above, I can see that the command being issued for my card is currently:

 http : PWD=/srv/http ; USER=root ; COMMAND=/usr/bin/amixer -M sset 100 16%

where it should (probably ?) be:

http : PWD=/srv/http ; USER=root ; COMMAND=/usr/bin/amixer -M sset 'A.Mstr Vol' 16%

where  'A.Mstr Vol' is the master volume simple control for my card as shown in:

RAConservatory:~  #  amixer
Simple mixer control 'A.Mstr Vol',0
  Capabilities: volume volume-joined
  Playback channels: Mono
  Capture channels: Mono
  Limits: 0 - 50
  Mono: 17 [34%] [-33.00dB]

If I use the revised command:

/usr/bin/amixer -M sset 'A.Mstr Vol' 25%

in an SSH session I can control the volume as needed (in this case raised to 25).

Chris

**Comment by findnix** [comment]


**Reply by rern** [reply]
Please try:
- Set Playback > show Volume then adjust volume knob
- Set Playback > hide Volume then adjust volume bar

A bug has been found in hidden Volume mode.

**Reply by findnix** [reply]
thanks ... but how can i change the volume when the volume dial is disabled?
btw: changing volume in the mpd-setup menu is working

**Reply by ceejayemm** [reply]
@rern 

Tried your request above. Made no difference to my setup, Volume always reverted to 100 both on volume knob and volume bar.

**Reply by rern** [reply]
@findnix,
Try:
```sh
wget https://github.com/rern/rAudio-1/raw/UPDATE/srv/http/assets/js/function.js -O /srv/http/assets/js/function.js
```
<br>

@ceejayemm,
Show your:
```sh
status.sh
mpd-data.sh
```

**Reply by findnix** [reply]
no success ...

rAudio:~  #  **status.sh**
{ "player" : "mpd" , "btclient" : false , "consume" : false , "control" : "Pre Box S2 Digital" , "counts" : { "album": 2, "albumartist": 3517, "artist": 6349, "composer": 2586, "date": 95, "genre": 456, "nas": 0, "sd": 0, "usb": 2, "song": 53614, "webradio": 5 } , "lcd" : false , "librandom" : false , "playlistlength" : 18 , "playlists" : 0 , "relays" : false , "relayson" : false , "updateaddons" : false , "volume" : 39 , "volumemute" : 0 , "webradio" : false , "repeat" : false , "random" : false , "single" : false , "elapsed" : false , "file" : "USB/1.42.6-25426/music/0000 Klassik [mp3+FLAC HD]/Vivaldi [FLAC]/Vivaldi in Venice [FLAC-192-24]/01 Vivaldi RV386 Movement 1 - Allegro.wav" , "song" : false , "state" : "stop" , "timestamp" : 1611058320920 , "updating_db" : false , "volumemute" : 0 , "Album" : "Vivaldi In Venice" , "Artist" : "Interpreti Veneizani" , "Time" : 279 , "Title" : "Vivaldi RV386 Movement 1 - Allegro" , "sampling" : "1/18 &bull; 24 bit 192.0 kHz 9.26 Mbit/s &bull; WAV" , "coverart" : "%2Fmnt%2FMPD%2FUSB%2F1.42.6-25426%2Fmusic%2F0000%20Klassik%20%5Bmp3%2BFLAC%20HD%5D%2FVivaldi%20%5BFLAC%5D%2FVivaldi%20in%20Venice%20%5BFLAC-192-24%5D%2Ffolder.1611058321.jpg"}

rAudio:~  #  **mpd-data.sh**
{ "devices" : [{ "aplayname" : "bcm2835 Headphones" , "card" : 0 , "device" : 0 , "dop" : 1 , "hw" : "hw:0,0" , "hwmixer" : "Headphone" , "mixers" : 1 , "mixerdevices" : ["Headphone"] , "mixermanual" : false , "mixertype" : "hardware" , "name" : "On-board Headphones" },{ "aplayname" : "Pre Box S2 Digital" , "card" : 1 , "device" : 0 , "dop" : 1 , "hw" : "hw:1,0" , "hwmixer" : "Pre Box S2 Digital " , "mixers" : 1 , "mixerdevices" : ["Pre Box S2 Digital "] , "mixermanual" : false , "mixertype" : "hardware" , "name" : "Pre Box S2 Digital" }] , "active" : true , "asoundcard" : 1 , "audioaplayname" : "Pre Box S2 Digital" , "audiooutput" : "Pre Box S2 Digital" , "autoupdate" : false , "buffer" : false , "bufferval" : false , "bufferoutput" : false , "bufferoutputval" : false , "counts" : { "album": 2, "albumartist": 3517, "artist": 6349, "composer": 2586, "date": 95, "genre": 456, "nas": 0, "sd": 0, "usb": 2, "song": 53614, "webradio": 5 } , "crossfade" : false , "crossfadeval" : false , "custom" : false , "ffmpeg" : true , "normalization" : false , "reboot" : "" , "replaygain" : false , "replaygainval" : "" , "soxr" : true , "soxrval" : "32 50 91.3 100 0 1" , "version" : "0.22.3-2" }
rAudio:~  #


please take a look at the  " " at the end of e.g hw mixer. 

rAudio:~  #  **amixer**
Simple mixer control 'Pre Box S2 Digital ',0
  Capabilities: pvolume pswitch
  Playback channels: Front Left - Front Right
  Limits: Playback 0 - 127
  Mono:
  Front Left: Playback 103 [81%] [-24.00dB] [on]
  Front Right: Playback 103 [81%] [-24.00dB] [on]
Simple mixer control 'Pre Box S2 Digital ',1
  Capabilities: pvolume pvolume-joined pswitch pswitch-joined
  Playback channels: Mono
  Limits: Playback 0 - 127
  Mono: Playback 127 [100%] [0.00dB] [on]

**Comment by findnix** [comment]
@rern
i misunderstood your commandline handling,
so my last comment and "findig" was wrong. i am sorry about that.
hence i deleted my last  comment.

a test with 
```
cmd.sh 'volume
10
40
Pre Box S2 Digital '
```
is changing the volume correctly, but the volume dial is still not working

**Comment by ceejayemm** [comment]
@rern 

As requested:

RAConservatory:~  #  status.sh
{ "player" : "mpd" , "btclient" : false , "consume" : false , "control" : "100" , "counts" : { "album": 313, "albumartist": 125, "artist": 559, "composer": 992, "date": 149, "genre": 33, "nas": 1, "sd": 0, "usb": 0, "song": 4771, "webradio": 20 } , "lcd" : false , "librandom" : false , "playlistlength" : 1 , "playlists" : 0 , "relays" : false , "relayson" : false , "updateaddons" : false , "volume" : 100 , "volumemute" : 0 , "repeat" : false , "random" : false , "single" : false , "elapsed" : false , "file" : "http://as-hls-uk-live.bbcfmt.s.llnwi.net/pool_904/live/uk/bbc_radio_two/bbc_radio_two.isml/bbc_radio_two-audio=320000.norewind.m3u8" , "song" : false , "state" : "stop" , "timestamp" : 1611058338345 , "updating_db" : false , "volumemute" : 0 , "Album" : "http://as-hls-uk-live.bbcfmt.s.llnwi.net/pool_904/live/uk/bbc_radio_two/bbc_radio_two.isml/bbc_radio_two-audio=320000.norewind.m3u8" , "Artist" : "BBC Radio 2" , "Name" : "" , "Time" : false , "Title" : "" , "webradio" : true , "sampling" : "1/1 &bull; " , "coverartradio" : "/data/webradiosimg/http:||as-hls-uk-live.bbcfmt.s.llnwi.net|pool_904|live|uk|bbc_radio_two|bbc_radio_two.isml|bbc_radio_two-audio=320000.norewind.m3u8.1611058338.jpg" , "coverart" : ""}
RAConservatory:~  #

RAConservatory:~  #  mpd-data.sh
{ "devices" : [{ "aplayname" : "merus-amp" , "card" : 0 , "device" : 0 , "dop" : 0 , "hw" : "hw:0,0" , "hwmixer" : "A.Mstr Vol" , "mixers" : 4 , "mixerdevices" : ["A.Mstr Vol","B.L Vol","C.R Vol","D.Lim thresh"] , "mixermanual" : false , "mixertype" : "hardware" , "name" : "MERUS Audio Amp" }] , "active" : true , "asoundcard" : 0 , "audioaplayname" : "merus-amp" , "audiooutput" : "MERUS Audio Amp" , "autoupdate" : false , "buffer" : false , "bufferval" : false , "bufferoutput" : false , "bufferoutputval" : false , "counts" : { "album": 313, "albumartist": 125, "artist": 559, "composer": 992, "date": 149, "genre": 33, "nas": 1, "sd": 0, "usb": 0, "song": 4771, "webradio": 20 } , "crossfade" : false , "crossfadeval" : false , "custom" : false , "ffmpeg" : true , "normalization" : false , "reboot" : "" , "replaygain" : false , "replaygainval" : "" , "soxr" : false , "soxrval" : "" , "version" : "0.22.3-2" }
RAConservatory:~  #

Chris

**Comment by findnix** [comment]
Good morning!

I just applied 20210120: 
On-board Headphone volume change works.
USB DAC does not. 
Same behavior as before: dial jumps back to the last successful set volume, change via system-> mpd -> mixer is OK

i am not sure if this journal entry is  is significant:

```
Jan 20 08:35:01 rAudio kernel: audit: type=1106 audit(1611128101.471:770): pid=4634 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:session_close grantors=pam_limits,pam_unix,pam_permit acct="root" exe="/usr/bin/sudo" hostname=? addr=? terminal=? res=success'
Jan 20 08:35:01 rAudio kernel: audit: type=1104 audit(1611128101.481:771): pid=4634 uid=33 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=pam_faillock,pam_permit,pam_env,pam_faillock acct="root" exe="/usr/bin/sudo" hostname=? addr=? terminal=? res=success'
Jan 20 08:35:01 rAudio nginx[346]: 2021/01/20 08:35:01 [error] 346#346: *216 FastCGI sent in stderr: "PHP message: PHP Notice:  Undefined variable: notverified in /srv/http/settings/addons.php on line 70PHP message: PHP Notice:  Undefined index: hide in /srv/http/settings/addons.php on line 51PHP message: PHP Warning:  exec(): Cannot execute a blank command in /srv/http/settings/addons.php on line 52PHP message: PHP Notice:  Undefined index: hide in /srv/http/settings/addons.php on line 51PHP message: PHP Warning:  exec(): Cannot execute a blank command in /srv/http/settings/addons.php on line 52PHP message: PHP Notice:  Undefined index: hide in /srv/http/settings/addons.php on line 51PHP message: PHP Warning:  exec(): Cannot execute a blank command in /srv/http/settings/addons.php on line 52PHP message: PHP Notice:  Undefined index: hide in /srv/http/settings/addons.php on line 51PHP message: PHP Warning:  exec(): Cannot execute a blank command in /srv/http/settings/addons.php on line 52PHP message: PHP Notice:  Undefined index: hide in /srv/http/settings/addons.php on line 51PHP message: PHP Warning:  exec(): Cannot execute a blank command in /srv/http/settings/addons.php on line 52" while reading response header from upstream, client: 192.168.0.178, server: localhost, request: "GET /settings/addons.php HTTP/1.1", upstream: "fastcgi://unix:/run/php-fpm/php-fpm.sock:", host: "192.168.0.218", referrer: "http://192.168.0.218/"

```

Ronald

**Reply by rern** [reply]
Post results of:
```sh
status.sh
mpd-data.sh
```

**Reply by findnix** [reply]
rAudio:~  #  **status.sh**

`{ "player" : "mpd" , "btclient" : false , "consume" : false , "control" : "Pre Box S2 Digital" , "counts" : { "album": 2, "albumartist": 3517, "artist": 6349, "composer": 2586, "date": 95, "genre": 456, "nas": 0, "sd": 0, "usb": 2, "song": 53614, "webradio": 5 } , "lcd" : false , "librandom" : false , "playlistlength" : 6 , "playlists" : 0 , "relays" : false , "relayson" : false , "updateaddons" : false , "volume" : 60 , "volumemute" : 0 , "webradio" : false , "repeat" : false , "random" : false , "single" : false , "elapsed" : 12 , "file" : "USB/1.42.6-25426/music/0000 Jazz [mp3+FLAC HD]/Eddie Harris [FLAC]/Eddie Harris - Swiss Movement (1969) [FLAC-44.1-16]/04 - You Got It in Your Soulness.flac" , "song" : 3 , "state" : "play" , "timestamp" : 1611142280149 , "updating_db" : false , "volumemute" : 0 , "Album" : "Swiss Movement" , "Artist" : "Les McCann & Eddie Harris" , "Time" : 455 , "Title" : "You Got It in Your Soulness" , "sampling" : "4/6 &bull; 16 bit 44.1 kHz 807 kbit/s &bull; FLAC" , "coverart" : "%2Fmnt%2FMPD%2FUSB%2F1.42.6-25426%2Fmusic%2F0000%20Jazz%20%5Bmp3%2BFLAC%20HD%5D%2FEddie%20Harris%20%5BFLAC%5D%2FEddie%20Harris%20-%20Swiss%20Movement%20%281969%29%20%5BFLAC-44.1-16%5D%2Ffolder.1611142280.jpg"}
`

rAudio:~  #  **mpd-data.sh**

`{ "devices" : [{ "aplayname" : "bcm2835 Headphones" , "card" : 0 , "device" : 0 , "dop" : 1 , "hw" : "hw:0,0" , "hwmixer" : "Headphone" , "mixers" : 1 , "mixerdevices" : ["Headphone"] , "mixermanual" : false , "mixertype" : "hardware" , "name" : "On-board Headphones" },{ "aplayname" : "Pre Box S2 Digital" , "card" : 1 , "device" : 0 , "dop" : 1 , "hw" : "hw:1,0" , "hwmixer" : "Pre Box S2 Digital " , "mixers" : 1 , "mixerdevices" : ["Pre Box S2 Digital "] , "mixermanual" : false , "mixertype" : "hardware" , "name" : "Pre Box S2 Digital" }] , "active" : true , "asoundcard" : 1 , "audioaplayname" : "Pre Box S2 Digital" , "audiooutput" : "Pre Box S2 Digital" , "autoupdate" : false , "buffer" : false , "bufferval" : false , "bufferoutput" : false , "bufferoutputval" : false , "counts" : { "album": 2, "albumartist": 3517, "artist": 6349, "composer": 2586, "date": 95, "genre": 456, "nas": 0, "sd": 0, "usb": 2, "song": 53614, "webradio": 5 } , "crossfade" : false , "crossfadeval" : false , "custom" : false , "ffmpeg" : true , "normalization" : false , "reboot" : "" , "replaygain" : false , "replaygainval" : "" , "soxr" : true , "soxrval" : "32 50 91.3 100 0 1" , "version" : "0.22.3-2" }
`

rAudio:~  #

**Reply by findnix** [reply]
Please take al look at the missing space at the end of "name" and "aplayname"

**mpd-data.sh**
```
rAudio:~  #  mpd-data.sh | jq '.devices[]'
{
  "aplayname": "bcm2835 Headphones",
  "card": 0,
  "device": 0,
  "dop": 1,
  "hw": "hw:0,0",
  "hwmixer": "Headphone",
  "mixers": 1,
  "mixerdevices": [
    "Headphone"
  ],
  "mixermanual": false,
  "mixertype": "hardware",
  "name": "On-board Headphones"
}
{
  "aplayname": "Pre Box S2 Digital",
  "card": 1,
  "device": 0,
  "dop": 1,
  "hw": "hw:1,0",
  "hwmixer": "Pre Box S2 Digital ",
  "mixers": 1,
  "mixerdevices": [
    "Pre Box S2 Digital "
  ],
  "mixermanual": false,
  "mixertype": "hardware",
  "name": "Pre Box S2 Digital"
}

```

**amixer**
```
rAudio:~  #  amixer
Simple mixer control 'Pre Box S2 Digital ',0
  Capabilities: pvolume pswitch
  Playback channels: Front Left - Front Right
  Limits: Playback 0 - 127
  Mono:
  Front Left: Playback 114 [90%] [-13.00dB] [on]
  Front Right: Playback 114 [90%] [-13.00dB] [on]
Simple mixer control 'Pre Box S2 Digital ',1
  Capabilities: pvolume pvolume-joined pswitch pswitch-joined
  Playback channels: Mono
  Limits: Playback 0 - 127
  Mono: Playback 127 [100%] [0.00dB] [on]
```

**aplay**
```
rAudio:~  #  aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: Headphones [bcm2835 Headphones], device 0: bcm2835 Headphones [bcm2835 Headphones]
  Subdevices: 8/8
  Subdevice #0: subdevice #0
  Subdevice #1: subdevice #1
  Subdevice #2: subdevice #2
  Subdevice #3: subdevice #3
  Subdevice #4: subdevice #4
  Subdevice #5: subdevice #5
  Subdevice #6: subdevice #6
  Subdevice #7: subdevice #7
card 1: Digital [Pre Box S2 Digital], device 0: USB Audio [USB Audio]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
rAudio:~  #

**Reply by rern** [reply]
Try these 2 revised files:
```sh
wget -O /srv/http/bash/cmd.sh https://github.com/rern/rAudio-1/raw/UPDATE/srv/http/bash/cmd.sh
wget -O /srv/http/bash/status.sh https://github.com/rern/rAudio-1/raw/UPDATE/srv/http/bash/status.sh
```

**Reply by findnix** [reply]
Looks very good. 
alsamixer shows the smooth changes directly  
I will test this evening and will report in detail tomorrow morning.

Thank You so match!
Ronald

**Reply by findnix** [reply]
@rern 

thank you for your high quality work and your quick response.
now everything runs smoothly.
i also did a "factory reset" last night. even after that, no problems with the volume control.

thanks again and greetings from austria
ronald

**Comment by ceejayemm** [comment, marked as answer]
@rern 

Just applied update 20210120, not seeing any errors like those reported by @findnix but my output device is a fixed HAT rather than USB. Volume control seems to work as expected, I have jumped between various screens / operations after setting various volume levels and selected volume levels have been maintained throughout.

Thanks for your efforts.

Chris
